src/AriesCounters.c

```diff
@@ -229,6 +255,9 @@ void FinalizeAriesCounters(MPI_Comm* mod16_comm, int my_rank, int reporting_rank
     // Allocate space to gather counters and timing.
     long long *counter_data = (long long*)malloc(sizeof(long long) * number_of_reporting_ranks * *AC_event_count);
     unsigned long long *timer_data = (unsigned long long *)malloc(sizeof(unsigned long long) * number_of_reporting_ranks);
+    struct timeval *start_times = (struct timeval *)malloc(sizeof(struct timeval) * number_of_reporting_ranks);
+    struct timeval *end_times = (struct timeval *)malloc(sizeof(struct timeval) * number_of_reporting_ranks);
```

Comment:
Now that I think of it, can't we just store *start_times and *end_times, and calculate *timer_data when writing out the file. It would also save us an MPI_Gather.